src/app/api/audit/summary/route.ts

```diff
@@ -25,14 +27,39 @@ export async function POST(request: NextRequest) {
       auth: account.access_token
     });
     const user = await octokit.rest.users.getAuthenticated();
-    const response = await octokit.rest.git.getTree({
+    const projectTree = await octokit.rest.git.getTree({
       owner: user.data.login,
       repo: repoName,
       tree_sha: 'main',
       recursive: 'true'
     });
 
-    return NextResponse.json({ data: response.data }, { status: response.status });
+    const allFiles = projectTree.data.tree
+      .filter((item) => item.type == 'file')
+      .map((file) => file.path);
+
+    const selectedFilesResponse = await fileAgent().invoke({
+      task: SUMMARY_SYSTEM_MESSAGE,
+      files: allFiles
+    });
+    const selectedFiles = fileAgentSchema.parse(selectedFilesResponse).files;
+
+    const selectedFilesContent = await Promise.all(
+      selectedFiles.map(async (file) => {
+        const fileContentResponse = await octokit.rest.repos.getContent({
+          owner: user.data.login,
+          repo: repoName,
+          path: file
+        });
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+        return Buffer.from((fileContentResponse.data as any).content, 'base64').toString('utf8');
+      })
```

Comment:
I really don't know how to avoid these, for some reason data.content is not recognized, even though it works as expected